Specs/Widgets/Viewer/ViewerSpec.js

```diff
@@ -1021,4 +1031,405 @@ defineSuite([
+        var tileset = new Cesium3DTileset({
+            url : path
+        });
+
+        // load the tileset then check tests
+        return tileset.readyPromise.then(function() {
+            var boundingSphere = tileset.boundingSphere;
+            var offset = new HeadingPitchRange(0.4, 1.2, 4 * boundingSphere.radius);
+
+            return viewer.zoomTo(tileset, offset).then(function() {
+                // moved to new location
+                expect(viewer.camera.position).not.toEqual(camPos);
+                expect(viewer.camera.direction).not.toEqual(camDir);
+            });
+        });
+    });
+
+    it('zoomTo zooms to entity with default offset when offset not defined', function() {
+        viewer = createViewer(container);
+        viewer.entities.add({
+            name : 'Blue box',
+            position: Cartesian3.fromDegrees(-114.0, 40.0, 300000.0),
+            box : {
+                dimensions : new Cartesian3(400000.0, 300000.0, 500000.0),
+                material : Color.BLUE
+            }
+        });
+
+        var entities = viewer.entities;
+
+        // stored for movement check
+        var camPos = Cartesian3.clone(viewer.camera.position);
+        var camDir = Cartesian3.clone(viewer.camera.direction);
+
+        return viewer.zoomTo(entities).then(function() {
+            // moved to new location
+            expect(viewer.camera.position).not.toEqual(camPos);
+            expect(viewer.camera.direction).not.toEqual(camDir);
+        });
+    });
+
+    it('zoomTo zooms to entity with offset', function() {
+        viewer = createViewer(container);
+        viewer.entities.add({
+            name : 'Blue box',
+            position: Cartesian3.fromDegrees(-114.0, 40.0, 300000.0),
+            box : {
+                dimensions : new Cartesian3(400000.0, 300000.0, 500000.0),
+                material : Color.BLUE
+            }
+        });
+
+        var entities = viewer.entities;
+
+        // stored for movement check
+        var camPos = Cartesian3.clone(viewer.camera.position);
+        var camDir = Cartesian3.clone(viewer.camera.direction);
+
+        var boundingSpheres = [];
+        var boundingSphereScratch = new BoundingSphere();
+        for (var i = 0, len = entities.length; i < len; i++) {
+            var state = viewer._dataSourceDisplay.getBoundingSphere(entities[i], false, boundingSphereScratch);
+
+            if (state === BoundingSphereState.PENDING) {
+                return;
+            } else if (state !== BoundingSphereState.FAILED) {
+                boundingSpheres.push(BoundingSphere.clone(boundingSphereScratch));
+            }
+        }
+        var boundingSphere = BoundingSphere.fromBoundingSpheres(boundingSpheres);
+        var offset = new HeadingPitchRange(3, 0.2, 2.3 * boundingSphere.radius);
+
+        return viewer.zoomTo(entities, offset).then(function() {
+            // moved to new location
+            expect(viewer.camera.position).not.toEqual(camPos);
```

Comment:
I think here and in the above tests, just spy on `camera.flyToBoundingSphere` or `viewBoundingSphere` and make sure it's being called with the options you expect it to be. You can also just pass a new hardcoded `HeadingPitchRange` with made up numbers since the function won't be called through. You don't need to iterate through all the bounding spheres in these tests